qctrlopencontrols/driven_controls/predefined.py

```diff
@@ -812,3 +813,129 @@ def _new_walsh_amplitude_modulated_filter_1_control(
         durations=durations,
         **kwargs,
     )
+
+
+def new_modulated_gaussian_control(
+    maximum_rabi_rate: float,
+    minimum_segment_duration: float,
+    duration: float,
+    modulation_frequency: float,
+) -> DrivenControl:
+    """
+    Generate a Gaussian driven control sequence modulated by a sinusoidal signal at a specific
+    frequency.
+
+    The net effect of this control sequence is an identity gate.
+
+    Parameters
+    ----------
+    maximum_rabi_rate: float
+        Maximum Rabi rate of the system.
+
+    minimum_segment_duration : float
+        Minimum length of each segment in the control sequence.
+
+    duration : float
+        Total duration of the control sequence.
+
+    modulation_frequency: float
+        Frequency of the modulation sinusoidal signal.
+
+    Returns
+    -------
+    DrivenControl
+        A control sequence as an instance of DrivenControl.
+    """
+
+    check_arguments(
+        maximum_rabi_rate > 0.0,
+        "Maximum Rabi rate must be greater than zero.",
+        {"maximum_rabi_rate": maximum_rabi_rate},
+    )
+
+    check_arguments(
+        minimum_segment_duration > 0.0,
+        "Minimum segment duration must be greater than zero.",
+        {"minimum_segment_duration": minimum_segment_duration},
+    )
+
+    check_arguments(
+        duration > minimum_segment_duration,
+        "Total duration must be greater than minimum segment duration.",
+        {"duration": duration, "minimum_segment_duration": minimum_segment_duration,},
+    )
+
+    # default spread of the gaussian shaped pulse as a fraction of its duration
+    _pulse_width = 0.1
+
+    # default mean of the gaussian shaped pulse as a fraction of its duration
+    _pulse_mean = 0.5
+
+    min_required_upper_bound = np.sqrt(2 * np.pi) / (_pulse_width * duration)
+    check_arguments(
+        maximum_rabi_rate >= min_required_upper_bound,
+        "Maximum Rabi rate must be large enough to permit a 2Pi rotation.",
+        {"maximum_rabi_rate": maximum_rabi_rate},
+        extras={
+            "minimum required value for upper_bound "
+            "(sqrt(2pi)/(0.1*maximum_duration))": min_required_upper_bound
+        },
+    )
+
+    segment_start_times = np.arange(0, duration, minimum_segment_duration)
+    segment_num = len(segment_start_times)
+    segment_midpoints = minimum_segment_duration / 2 + segment_start_times
+
+    # prepare the modulation signals. We use sinusoids that are zero at the center of the pulse,
+    # which ensures the pulses are antisymmetric about the center of the pulse and thus effect a net
+    # zero rotation.
+    modulation_signals = np.sin(
```

Comment:
nit: how about we move this inside the `modulation_frequency != 0` block?

Reply:
SG, as it's only used in that case.